ta/pkcs11/src/pkcs11_helpers.h

```diff
@@ -55,6 +61,13 @@ bool key_type_is_symm_key(uint32_t key_type_id);
 /* Return true if the key type @key_type_id relates to an asymmetric key */
 bool key_type_is_asymm_key(uint32_t key_type_id);
 
+/* Calculate AES key check value size */
+enum pkcs11_rc aes_get_check_value_size(void *key, uint32_t key_size,
+					uint32_t *kcv_size);
+/* Calculate AES key check value data */
+enum pkcs11_rc aes_compute_check_value(void *key, uint32_t key_size,
+				       void *kcv, uint32_t *kcv_size);
```

Comment:
This function is only called from `check_attr_checksum()`, please move it in the related source files as a static function.

The function applies to CKK_AES, CKK_DES and CKK_DES3. I suggest to rename the static function `compute_check_value_with_ecb()` and pass the key_type ID. It would currently support only PKCS11_CKK_AES, but ill be easy to later extend to CKK_DES and CKK_DES3 that are computed the same way.

CKA_CHECK_VALUE attribute is always 3 bytes so need to pass kcv_size as argument. This function could get instead a `struct obj_attrs` as argument, preallocated by caller (`sizeof(struct obj_attrs) + PKCS11_CKA_CHECK_VALUE_SIZE)` and fill it with the computed value.